rdflib/plugins/stores/memory.py

```diff
@@ -470,11 +474,10 @@ def __add_triple_context(self, triple, context, quoted):
 
         # if this is the first ever triple in the store, set default ctx info
         if self.__defaultContexts is None:
-            self.__defaultContexts = self.__tripleContexts[triple]
-
+            self.__defaultContexts = triple_context
         # if the context info is the same as default, no need to store it
-        if self.__tripleContexts[triple] == self.__defaultContexts:
-            del self.__tripleContexts[triple]
+        if triple_context == self.__defaultContexts:
+            del triple_context
```

Comment:
Shouldn't this delete the context from `self.__tripleContexts`? It appears to me, that `del triple_context` just does nothing.

Reply:
Yes, it strongly seems you are right.
```
>>> x={"a","b"}
>>> y=x
>>> del y``
>>> x``
set(['a', 'b'])``
>>> y``
Traceback (most recent call last):``
File "<stdin>", line 1, in <module>``
NameError: name 'y' is not defined``
```

It should probably be:
```
del self.__tripleContexts[triple]
```

Reply:
I did the fix in eb5e025177ad36bd0cd7ac6a794cbc42943826f1

Reply:
Yes you're right, thanks @white-gecko